apps/worker/src/request_processor/dsnp.announcement.processor.ts

```diff
@@ -0,0 +1,338 @@
+            asset.references?.forEach(async (reference) => {
+              const contentBuffer = await this.ipfsService.getPinned(reference.referenceId);
+              const hashedContent = await calculateDsnpHash(contentBuffer);
+              const video: ActivityContentVideoLink = {
+                mediaType: reference.mimeType,
+                hash: [hashedContent],
+                height: reference.height,
+                width: reference.width,
+                type: 'Link',
+                href: await this.formIpfsUrl(reference.referenceId),
+              };
+              duration = duration ?? reference.duration ?? '';
+              videoLinks.push(video);
+            });
+            const videoActivity: ActivityContentVideo = {
+              type: 'Video',
+              name: asset.name || '',
+              url: videoLinks,
+              duration,
+            };
+
+            attachments.push(videoActivity);
+            break;
+          }
+          case AttachmentTypeDto.AUDIO: {
+            const audioLinks: ActivityContentAudioLink[] = [];
+            let duration = '';
+            asset.references?.forEach(async (reference) => {
+              const contentBuffer = await this.ipfsService.getPinned(reference.referenceId);
+              const hashedContent = await calculateDsnpHash(contentBuffer);
+              duration = duration ?? reference.duration ?? '';
+              const audio: ActivityContentAudioLink = {
+                mediaType: reference.mimeType,
+                hash: [hashedContent],
+                type: 'Link',
+                href: await this.formIpfsUrl(reference.referenceId),
+              };
+              audioLinks.push(audio);
+            });
+            const audioActivity: ActivityContentAudio = {
+              type: 'Audio',
+              name: asset.name || '',
+              url: audioLinks,
+              duration,
+            };
+
+            attachments.push(audioActivity);
+            break;
+          }
+          default:
+            throw new Error(`Unsupported attachment type ${typeof asset.type}`);
+        }
+      });
+    }
+
+    const note = createNote(noteContent?.content.content ?? '', new Date(noteContent?.content.published ?? ''), {
+      name: noteContent?.content.name,
+      location: {
+        latitude: noteContent?.content.location?.latitude,
+        longitude: noteContent?.content.location?.longitude,
+        radius: noteContent?.content.location?.radius,
+        altitude: noteContent?.content.location?.altitude,
+        accuracy: noteContent?.content.location?.accuracy,
+        name: noteContent?.content.location?.name || '',
+        type: 'Place',
+      },
+      tag: tags,
+      attachment: attachments,
+    });
+    const noteString = JSON.stringify(note);
+    const [cid, hash] = await this.pinBufferToIPFS(Buffer.from(noteString));
+    const ipfsUrl = await this.formIpfsUrl(cid);
+    return [cid, hash, ipfsUrl];
```

Comment:
the order is not correct in here you are returning `[cid, hash, ipfsUrl]` and then in processBroadcast and other usages you are mapping to `[cid, ipfsUrl, hash]`